backend/user-service/src/main/java/hu/me/iit/malus/thesis/user/service/impl/UserServiceImpl.java

```diff
@@ -283,11 +212,11 @@ public void saveCourseAssign(String studentEmail, Long courseId) {
      * {@inheritDoc}
      */
     @Override
-    public Student getStudentByEmail(String studentEmail) {
+    public StudentDto getStudentByEmail(String studentEmail) {
         try {
             Optional<Student> optionalStudent = studentRepository.findByEmail(studentEmail);
-            if (!optionalStudent.isPresent()) throw new UserNotFoundException(studentEmail);
-            return optionalStudent.get();
+            if (optionalStudent.isEmpty()) throw new UserNotFoundException(studentEmail);
```

Comment:
igy is jo, de nezz ra erre: https://www.baeldung.com/java-optional

Reply:
rendben, majd a jövőben akkor ez a orElseGet/Throw-ot fogom használni.